src/components/Timer/useTimer.ts

```diff
@@ -0,0 +1,61 @@
+import moment from "moment"
+import { Duration } from "moment"
+import { useEffect, useRef, useState } from "react"
+
+
+export type UseTimerOptions = { minutes: number, seconds: number }
+export const useTimer = ({
+    minutes, seconds
+}: UseTimerOptions) => {
+
+    const [timer, setTimer] = useState<Duration>(moment.duration({ minutes, seconds }))
```

Comment:
Moment makes dealing with the timer a lot easier. No need to parse strings or process values manually